.github/workflows/golangci-lint.yaml

```diff
@@ -0,0 +1,19 @@
+name: golangci-lint
+
+# Switch to only manual triggering for now.
+on:
+  workflow_dispatch:
```

Comment:
Run thhis on PR just like the E2E tests as well.
```
on:
  push:
    branches:
      - main
  pull_request:
    branches:
      - main  
```

we don't need the manual trigger.

Reply:
That's a lot of linting issues 😮 ! I'm fine with leaving it as is right now. We'd need to revisit in future. Things like the `errcheck` are very important because it'll help us catch and log errors